sql_grader/mixins/grading.py

```diff
@@ -1,21 +1,72 @@
 """
 Mixin grading-related functionality
 """
+import logging
+import os.path
+from xblock.fields import Boolean
 from xblock.fields import Float
 from xblock.fields import Integer
 from xblock.fields import Scope
+from xblock.fields import String
+
 from xblock.scorable import ScorableXBlockMixin
 from xblock.scorable import Score
 
-from ..models import SqlProblem
-from ..models import DATABASES
+from ..models import all_datasets
+
+try:
+    from django.utils.translation import ugettext_lazy as _
+except ImportError:
+    def _(text):
+        """
+        Mock translation for scraping
+        """
+        return text
 
+log = logging.getLogger('sql_grader')
 
-def _(text):
+
+# pylint: disable=unused-argument
+def attempt_safe(dataset, answer_query, verify_query, is_ordered, query):
     """
-    Mock translation for scraping
+    Attempt a SqlProblem, using codejail to sandbox the execution.
     """
-    return text
+    # pylint: disable=E0401,C0415
+    from capa.safe_exec import safe_exec
+    from codejail.safe_exec import SafeExecException
```

Comment:
Can we add codejail as an explicit dependency and copy the `safe_exec` code out of capa? I know that there's a limit to how much dev time we want to put into this, but this strikes me as one of those obscure XBlocks that will quietly go doing it's thing for years without anyone knowing or touching it. Until the day comes when we change how codejail works in platform or move capa somewhere else, at which point it'll start breaking for unclear reasons.

Reply:
This could just use the `safe_exec` [out of codejail](https://github.com/edx/codejail/blob/master/codejail/safe_exec.py). Capa's version of `safe_exec` adds module imports for things that might be used by capa problems.

Reply:
I don't know why codejail isn't published to pypi. This makes things more inconvenient.